packages/caravan-health/src/privacy.ts

```diff
@@ -0,0 +1,285 @@
+
+    const amount = utxo.utxos.reduce((sum, utxo) => sum + utxo.value, 0);
+    if (reused) {
+      reusedAmount += amount;
+    }
+    totalAmount += amount;
+  }
+
+  return reusedAmount / totalAmount;
+}
+
+async function isReusedAddress(
+  address: string,
+  client: BlockchainClient,
+): Promise<boolean> {
+  let txs: Array<Transaction> = await client.getAddressTransactions(address);
+  if (txs.length > 1) {
+    return true;
+  }
+  return false;
+}
+/*
+If we are making payment to other wallet types then the privacy score should decrease because 
+the change received will be to an address type matching our wallet and it will lead to a deduction that
+we still own that amount.
+*/
+export function addressTypeFactor(
+  transactions: Array<Transaction>,
+  walletAddressType: string,
+): number {
+  let P2WSH: number = 0;
+  let P2PKH: number = 0;
+  let P2SH: number = 0;
+  let atf: number = 1;
+  transactions.forEach((tx) => {
+    tx.vout.forEach((output) => {
+      let address = output.scriptPubkeyAddress;
+      if (address.startsWith("bc")) {
+        //Bech 32 Native Segwit (P2WSH) or Taproot
+        P2WSH += 1;
+      } else if (address.startsWith("1")) {
+        // Legacy (P2PKH)
+        P2PKH += 1;
+      } else {
+        // Segwith (P2SH)
+        P2SH += 1;
+      }
+    });
+  });
+
+  if (walletAddressType == "P2WSH" && P2WSH != 0 && (P2SH != 0 || P2PKH != 0)) {
+    atf = 1 / (P2WSH + 1);
+  } else if (
+    walletAddressType == "P2PKH" &&
+    P2PKH != 0 &&
+    (P2SH != 0 || P2WSH != 0)
+  ) {
+    atf = 1 / (P2PKH + 1);
+  } else if (
+    walletAddressType == "P2SH" &&
+    P2SH != 0 &&
+    (P2WSH != 0 || P2PKH != 0)
+  ) {
+    atf = 1 / (P2SH + 1);
+  } else {
+    atf = 1;
+  }
+  return atf;
+}
+
+/* 
+The spread factor using standard deviation helps in assessing the dispersion of UTXO values.
+In Bitcoin privacy, spreading UTXOs reduces traceability by making it harder for adversaries
+to link transactions and deduce the ownership and spending patterns of users.
+*/
+export function utxoSpreadFactor(utxos: Array<WalletUTXOs>): number {
+  const amounts: Array<number> = utxos.map((utxo) =>
+    utxo.utxos.reduce((sum, utxo) => sum + utxo.value, 0),
+  );
+  const mean: number =
+    amounts.reduce((sum, amount) => sum + amount, 0) / amounts.length;
+  const variance: number =
+    amounts.reduce((sum, amount) => sum + Math.pow(amount - mean, 2), 0) /
+    amounts.length;
+  const stdDev: number = Math.sqrt(variance);
+  return stdDev / (stdDev + 1);
+}
+
+/* 
+The weightage is ad-hoc to normalize the privacy score based on the number of UTXOs in the set.
+- 0 for UTXO set length >= 50
+- 0.25 for UTXO set length >= 25 and <= 49
+- 0.5 for UTXO set length >= 15 and <= 24
+- 0.75 for UTXO set length >= 5 and <= 14
+- 1 for UTXO set length < 5
+*/
+export function utxoSetLengthWeight(utxos: Array<WalletUTXOs>): number {
+  let utxoSetLength = utxos.reduce((sum, utxo) => sum + utxo.utxos.length, 0);
```

Comment:
Something doesn't feel right about this typing. a utxo shouldn't have a nested list of utxos imo. Seems like maybe this function is expecting a list of transactions and then is getting the number of outputs for each transaction?